src/core/src/pass/serialize.cpp

```diff
@@ -965,11 +966,35 @@ void serialize_rt_info(pugi::xml_node& root, const std::string& name, const ov::
             serialize_rt_info(child, it.first, it.second);
         }
     } else {
-        std::string value = data.as<std::string>();
+        const auto& value = data.as<std::string>();
         child.append_attribute("value").set_value(value.c_str());
     }
 }
 
+bool append_custom_rt_info(pugi::xml_node& node, const std::string& name, const ov::Any& data) {
+    if (std::regex_search(name, std::regex{"^__"}))
+        return false;
+
+    auto custom_node = node.append_child("custom");
+    custom_node.append_attribute("name").set_value(name.c_str());
+    bool appended = false;
+
+    if (data.is<ov::AnyMap>()) {
+        const auto& any_map = data.as<ov::AnyMap>();
+        for (const auto& it : any_map)
+            appended = append_custom_rt_info(custom_node, it.first, it.second) || appended;
+
+    } else if (!data.empty() && !data.is<ov::RuntimeAttribute>() && !data.is<std::shared_ptr<ov::RuntimeAttribute>>()) {
+        const auto& value = data.as<std::string>();
+        custom_node.append_attribute("value").set_value(value.c_str());
+        appended = true;
+    }
+
+    if (!appended)
+        node.remove_child(custom_node);
```

Comment:
Move it inside `(data.is<ov::AnyMap>())`
and appended true is not only if AnyMap is not empty?

Reply:
Resolving w/o code update: As discussed offline current solution makes the code most direct: nested `AnyMap` entries w/o recurrence would be code bloat, while `remove_child(custom_node)` is rare and not relevant from performance perspective.